### PR TITLE
Refactor to handle dob

### PR DESCRIPTION
**Pull request by money**
Made updates to handle more options for querying a client (i.e. suffix and dob).

### CODE REVIEW

File: spec/helpers/login_helper.rb [hunk 1]
```diff
@@ -49,7 +49,7 @@ def multi_factor_auth
 def mfa_page?
   sleep ENV.fetch('MAX_WAIT', Capybara.default_max_wait_time).to_i
   Wait.for_document
-  Capybara.current_url.include?('login')
+  Capybara.current_url.include?('cognito')
 end
```
Comment (NagaPradeepGajula-osi) [inline spec/helpers/login_helper.rb]: I think we should have both versions of the url and code here for pre-int, Int and staging as separate based on the environment.

File: spec/snapshot/state_sensitive_user_spec.rb [hunk 2]
```diff
@@ -19,81 +19,95 @@
...
-          expect(page).to have_css('span.person', text: same_county_sensitive_not_sealed.full_name)
+          expect(page).to \
+            have_css('span.person', text: same_county_sensitive_not_sealed.full_name)
         end
       end
 
       it 'should be able to attach different_county_sensitive_not_sealed client' do
-        search_client(query: different_county_sensitive_not_sealed.search_name)
+        search_client(query: different_county_sensitive_not_sealed.search_name_with_dob)
+        byebug
```
Comment (NagaPradeepGajula-osi) [inline spec/snapshot/state_sensitive_user_spec.rb]: delete it.

File: spec/helpers/login_helper.rb (hunk 1 above)
Comment (money) [reply spec/helpers/login_helper.rb]: you are correct.  will make the change

File: spec/snapshot/state_sensitive_user_spec.rb (hunk 2 above)
Comment (money) [reply spec/snapshot/state_sensitive_user_spec.rb]: done

File: spec/helpers/login_helper.rb [hunk 3]
```diff
@@ -38,7 +38,8 @@ def low_level_environment(user)
 
 def multi_factor_auth
   return unless mfa_page?
-  Capybara.fill_in('Verification Code', with: ENV['VERIFICATION_CODE'])
+
+  Capybara.fill_in('Code', with: ENV['MFA'])
   if %i[selenium_ie selenium_edge].include?(Capybara.current_driver)
```
Comment (NagaPradeepGajula-osi) [inline spec/helpers/login_helper.rb]: Same goes for this as the below

File: lib/person.rb [hunk 4]
```diff
@@ -31,10 +31,19 @@ def assigns_role(args)
   end
 
   def full_name
-    [first_name, middle_name, last_name].reject(&:blank?).join(' ')
+    name = [first_name, middle_name, last_name].reject(&:blank?).join(' ')
+    [name, suffix].reject(&:blank?).join(', ')
   end
 
   def search_name
```
Comment (pokharelmanish) [inline lib/person.rb]: The method `search_name`, `full_search`, `search_name_with_dob` have similar code. Probably we can used one method and pass `full_name`, `last_name`, `date_of_birth`, `ssn` as parameters. Something like this. DRY

```
def search(full_name, date_of_birth = '', ssn ='') 
  [full_name, date_of_birth, ssn].reject(&:blank?).join(' ')
end
```

Comment (money) [reply lib/person.rb]: There is going to be some refactor work where this will be addressed.

**Pull request merged**